memcpy regression test fails on OS X

The `test/basic/test_memcpy.c` regression fails (with `FAILED` result) on OS X.

The generated compiler warnings provide a hint:
````
smack test/basic/test_memcpy.c
SMACK program verifier version 1.8.0
test/basic/test_memcpy.c:9:3: warning: implicit declaration of function '__builtinx_object_size' is invalid in C99 [-Wimplicit-function-declaration]
  memcpy(&a, &b, sizeof(int));
  ^
/usr/include/secure/_string.h:65:43: note: expanded from macro 'memcpy'
  __builtin___memcpy_chk (dest, src, len, __darwin_obsz0 (dest))
                                          ^
/usr/include/secure/_common.h:38:32: note: expanded from macro '__darwin_obsz0'
#define __darwin_obsz0(object) __builtin_object_size (object, 0)
                               ^
/usr/local/share/smack/include/smack.h:21:31: note: expanded from macro '__builtin_object_size'
#define __builtin_object_size __builtinx_object_size
                              ^
1 warning generated.
/usr/local/share/smack/lib/smack.c(40,1): This assertion can fail
Execution trace:
    /usr/local/share/smack/lib/smack.c(2231,3): Trace: Thread=1  
...
SMACK found an error.
````
This seems to be happening since `memcpy` is a macro which gets translated to `__memcpy_chk`. The generated Boogie code thus calls this `__memcpy_chk` function for which we do not have a stub.
````
procedure {:entrypoint} main()
  returns ($r: i32)
{
   ...
  assume {:sourceloc "test/basic/test_memcpy.c", 9, 3} true;
  call $i6 := __builtinx_object_size.ref.i32($p0, 0);
  assume {:sourceloc "test/basic/test_memcpy.c", 9, 3} true;
  $i7 := $sext.i32.i64($i6);
  assume {:sourceloc "test/basic/test_memcpy.c", 9, 3} true;
  call $p8 := __memcpy_chk($p4, $p5, 4, $i7);
  assume {:sourceloc "test/basic/test_memcpy.c", 10, 3} true;
  $i9 := $load.i32($M.0, $p0);
  assume {:sourceloc "test/basic/test_memcpy.c", 10, 3} true;
  $i10 := $load.i32($M.0, $p1);
  assume {:sourceloc "test/basic/test_memcpy.c", 10, 3} true;
  $i11 := $eq.i32($i9, $i10);
  assume {:sourceloc "test/basic/test_memcpy.c", 10, 3} true;
  $i12 := $zext.i1.i32($i11);
  assume {:sourceloc "test/basic/test_memcpy.c", 10, 3} true;
  call __VERIFIER_assert($i12);
  assume {:sourceloc "test/basic/test_memcpy.c", 11, 3} true;
  $r := 0;
  $exn := false;
  return;
}
````

--- a/share/smack/include/smack.h
+++ b/share/smack/include/smack.h
@@ -13,6 +13,7 @@
 extern "C" {
 #endif
 
+#ifdef SVCOMP
 // Apprently needed by SVCOMP benchmarks
 #define __inline
 #define __builtin_expect __builtinx_expect
@@ -22,6 +23,7 @@ extern "C" {
 
 // For handling of va_start macro
 void __builtinx_va_start(char*,char*);
+#endif
 
 void __SMACK_code(const char *fmt, ...);
 void __SMACK_mod(const char *fmt, ...);

--- a/share/smack/svcomp/utils.py
+++ b/share/smack/svcomp/utils.py
@@ -49,6 +49,7 @@ def svcomp_frontend(args):
   name, ext = os.path.splitext(os.path.basename(args.input_files[0]))
   svcomp_process_file(args, name, ext)
 
+  args.clang_options += " -DSVCOMP"
   args.clang_options += " -DAVOID_NAME_CONFLICTS"
   args.clang_options += " -DCUSTOM_VERIFIER_ASSERT"
   args.clang_options += " -DNO_FORALL"

